allennlp/state_machines/transition_functions/linking_transition_function.py

```diff
@@ -96,11 +96,19 @@ def _compute_action_probabilities(self,
         for group_index in range(group_size):
             instance_actions = actions[group_index]
             predicted_action_embedding = predicted_action_embeddings[group_index]
-            action_embeddings, output_action_embeddings, embedded_actions = instance_actions['global']
-            # This is just a matrix product between a (num_actions, embedding_dim) matrix and an
-            # (embedding_dim, 1) matrix.
-            embedded_action_logits = action_embeddings.mm(predicted_action_embedding.unsqueeze(-1)).squeeze(-1)
-            action_ids = embedded_actions
+            embedded_actions: List[int] = []
+
+            output_action_embeddings = predicted_action_embeddings.new_tensor([], dtype=torch.float)
```

Comment:
I don't think you actually need to construct these tensors.  Just have whatever variables are needed here be `None`, and assign them appropriately below depending on the path the logic takes.